### PR TITLE
py-qdarkstyle: fix dependencies & add Python 3.8

### PR DESCRIPTION
**Pull request by MarcusCalhoun-Lopez**
###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.3
Xcode 11.3.1 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by MarcusCalhoun-Lopez** [comment]
See [utils/scss.py](https://github.com/ColinDuquesnoy/QDarkStyleSheet/blob/2dc4ab52a70c9ebd66f1e0afa1a551a73565db81/qdarkstyle/utils/scss.py#L14) for dependency on py-qtsass.

File: python/py-qdarkstyle/Portfile [hunk 1]
```diff
@@ -24,13 +24,16 @@ checksums           sha256  6a967c4b664446f8bed9df12d1032cf68cb54f186bfc9cbfdbbc
...
 
 if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
     depends_lib-append \
                     port:py${python.version}-helpdev \
                     port:py${python.version}-qtpy \
-                    port:py${python.version}-setuptools
+                    port:py${python.version}-qtsass
```
Comment (reneeotten) [inline python/py-qdarkstyle/Portfile]: I don't agree with the change in dependency type. The port installs a ```console_script```, which uses ```pkg_resources``` from ```py-setuptools``` and it is, therefore, also required at runtime. 

Secondly, the ```py-qtsass``` is listed as under ```extra_requires``` in the category "develop", which I typically wouldn't add. However, I do  agree that the unconditional import in the file you mention would suggest it should not be listed as an optional dependency... If you do include this dependency, what is the rationale for not adding ```watchdog```, which is also imported in ```scripts/process_qrc.py```?

File: python/py-qdarkstyle/Portfile [hunk 2]
```diff
@@ -24,13 +24,16 @@ checksums           sha256  6a967c4b664446f8bed9df12d1032cf68cb54f186bfc9cbfdbbc
                     rmd160  95b7cbcdbec958e2ee685aff8348e5601a2a1c35 \
                     size    405579
 
-python.versions     27 35 36 37
+python.versions     27 35 36 37 38
```
Comment (reneeotten) [inline python/py-qdarkstyle/Portfile]: upstream does not declare compatibility with Python 3.8; I assume you verified it is in fact working correctly?

File: python/py-qdarkstyle/Portfile (hunk 1 above)
Comment (MarcusCalhoun-Lopez) [reply python/py-qdarkstyle/Portfile]: > The port installs a `console_script`, which uses `pkg_resources` from `py-setuptools` and it is, therefore, also required at runtime.

Please forgive my ignorance, but would you mind telling me where the `console_script` is?
I am not disagreeing with your assessment, but I am unable to find the installed `console_script`, and I would like to expand my own knowledge.

> [W]hat is the rationale for not adding watchdog, which is also imported in scripts/process_qrc.py?

Again, I do not mean to be obtuse, but I cannot find a file called `process_qrc.py`.

File: python/py-qdarkstyle/Portfile (hunk 2 above)
Comment (MarcusCalhoun-Lopez) [reply python/py-qdarkstyle/Portfile]: I was able to successfully run the [example script](https://github.com/ColinDuquesnoy/QDarkStyleSheet/blob/2dc4ab52a70c9ebd66f1e0afa1a551a73565db81/example/example.py).

File: python/py-qdarkstyle/Portfile (hunk 1 above)
Comment (reneeotten) [reply python/py-qdarkstyle/Portfile]: ah, sorry you are right about the ```process_qrc.py``` file, which is present in the GitHub repository, but apparently does not get installed. So then it should be fine to only add ```py-qtsass```. 

Regarding the console_script, it's this line in the setup.py:
```    entry_points={"console_scripts": ["qdarkstyle=qdarkstyle.__main__:main"]},```
which installs for py37-qdarkstyle the file ```/opt/local/bin/qdarkstyle-3.7```. If you look at its contents you'll notice that it does ```from pkg_resources import load_entry_point```, which will require ```py-setuptools``` at runtime.

File: python/py-qdarkstyle/Portfile (hunk 2 above)
Comment (reneeotten) [reply python/py-qdarkstyle/Portfile]: alright, that's should do it - thanks!

File: python/py-qdarkstyle/Portfile (hunk 1 above)
Comment (MarcusCalhoun-Lopez) [reply python/py-qdarkstyle/Portfile]: Thank you for taking the time for the addition explanation.

Comment (petrrr) [reply python/py-qdarkstyle/Portfile]: Hi, if this really is a hard dependency, (1) I would suggest to have this fixed upstream and (2) just add it as `depends_lib`.

**Comment by MarcusCalhoun-Lopez** [comment]
1. I am not sure what exactly the upstream fix would be.
1. As requested, the dependency type is now `depends_lib`.

**Pull request merged**